app/src/main/java/com/arygm/quickfix/ui/dashboard/AnnouncementsWidget.kt

```diff
@@ -0,0 +1,245 @@
+    val horizontalSpacing = maxWidth * 0.025f // 2.5% of the available width for spacing
+
+    Column(
+        modifier =
+            modifier
+                .fillMaxWidth()
+                .padding(horizontalSpacing)
+                .shadow(5.dp, RoundedCornerShape(20.dp))
+                .background(MaterialTheme.colorScheme.surface, RoundedCornerShape(12.dp))
+                .testTag("AnnouncementsWidget")) {
+          // Header with Show All button
+          Row(
+              modifier = Modifier.fillMaxWidth().padding(8.dp),
+              horizontalArrangement = Arrangement.SpaceBetween,
+              verticalAlignment = Alignment.CenterVertically) {
+                Text(
+                    text = "Announcements",
+                    color = MaterialTheme.colorScheme.onBackground,
+                    style = poppinsTypography.headlineMedium,
+                    fontSize = 19.sp,
+                    modifier = Modifier.testTag("AnnouncementsTitle"))
+                TextButton(
+                    onClick = { showAll = !showAll },
+                    modifier = Modifier.testTag("ShowAllButton")) {
+                      Text(
+                          text = if (showAll) "Show Less" else "Show All",
+                          color = MaterialTheme.colorScheme.onSurface,
+                          style = MaterialTheme.typography.bodyMedium,
+                          fontWeight = FontWeight.SemiBold)
+                    }
+              }
+
+          Divider(
+              color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.2f),
+              thickness = 1.dp,
+              modifier = Modifier.testTag("AnnouncementsDivider"))
+
+          val itemsToShow = if (showAll) announcements else announcements.take(itemsToShowDefault)
+          itemsToShow.forEachIndexed { index, announcement ->
+            val pairs = imagesForAnnouncements[announcement.announcementId] ?: emptyList()
+            val bitmapToDisplay = pairs.firstOrNull()?.second
+            var category by remember { mutableStateOf(Category()) }
+            LaunchedEffect(Unit) {
+              categoryViewModel.getCategoryBySubcategoryId(
+                  announcement.category,
+                  onSuccess = {
+                    if (it != null) {
+                      category = it
+                    }
+                  })
+            }
+
+            AnnouncementItem(
+                announcement = announcement,
+                announcementImage = bitmapToDisplay,
+                categoryIcon = getCategoryIcon(category),
+                onClick = {
+                  announcementViewModel.selectAnnouncement(announcement)
+                  navigationActions.navigateTo(UserScreen.ANNOUNCEMENT_DETAIL)
+                })
+
+            if (index < itemsToShow.size - 1) {
+              Divider(
+                  color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.2f),
```

Comment:
could import colorScheme to rather have `colorScheme.onSurface.copy(alpha = 0.2f)`